MO: Fix Php Warning: Round() Expects Parameter 2 To Be Long

firstred opened the pull request:
Solves `Php Warning: Round() Expects Parameter 2 To Be Long` because `_PS_PRICE_COMPUTE_PRECISION_` isn't defined on all versions of PrestaShop.

Defaults to `0` if `_PS_PRICE_COMPUTE_PRECISION_` is not defined.

Also converted to PSR-2 + PHPDoc comments

firstred commented on socialsharing.php:
```diff
@@ -1,277 +1,375 @@
+                $this->getCacheId('socialsharing_he ader|'.
+                    (isset($product->id) && $product->id ? (int)$product->id : '')))
+            ) {
+                if (!defined('_PS_PRICE_COMPUTE_PRECISION_')) {
```
`_PS_PRICE_COMPUTE_PRECISION_` is not defined on PS versions < 1.6.0.11

mickaelandrieu replied:
... but if shops are updated to 1.6.1.5, this constant is defined so ... we dont need to do a check.

The only - and very good - solution is to update version of PrestaShop in concerned websites, right ?

firstred replied:
Yep, release early, release often, totally agree! But don't forget that these modules are also pushed to older versions.

I'm afraid we also can't ignore the the devs who are forced to make their modules compatible with older versions :)
This is the first thing I see (with `xdebug`) whenever I fire up a new instance of the first stable 1.6 version:
![1605socialsharin](https://cloud.githubusercontent.com/assets/6775736/14590399/b3e7a48e-04f9-11e6-9ca7-cd698f96e239.png)

Would be worth spilling a few CPU cycles on newer PS versions, right?

mickaelandrieu replied:
Yes,  let's go for it! 

:+1:

firstred commented:
Rebased and changed into an atomic commit ( was huge :open_mouth: )
@mickaelandrieu Still going for it? :stuck_out_tongue_winking_eye: 
ping @Shudrum @maximebiloe @julienbourdeau @vincentbz @xBorderie

mickaelandrieu commented:
LGTM, thanks!
